dockerfiles/ws-health-exporter/exporter.py

```diff
@@ -2,54 +2,157 @@
 # -*- coding: utf-8 -*-
 
 import sys
-import os
 import logging
 import traceback
 import json
+import io
+from urllib.parse import urlparse
+from threading import Lock
 
 from flask import Flask
+from waitress import serve
 from apscheduler.schedulers.background import BackgroundScheduler
 from prometheus_client import generate_latest, Gauge
 from websocket import create_connection
+from environs import Env
 
 LOGGING_FORMAT = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
 
-PORT = os.environ.get("PORT", "8001")
-WS_NODE_URL = os.environ.get("NODE_URL", "ws://127.0.0.1:5556")
-WS_TIMEOUT = 60
 ws_metrics = {
-    'polkadot_ws_alive': Gauge('wss_alive', 'WebSocket alive', [])
+    'polkadot_ws_alive': Gauge('wss_alive', 'WebSocket alive', ['url'])
 }
-# global variable to keep current ws status
-readiness_status = False
+
+# global variable to keep current readiness status
+# we need threading.Lock() to avoid race conditions
+# some Python modules can use threads inside
+readiness_status = {'status': False, 'lock': Lock()}
+
+# Flask app
 app = Flask(__name__, static_folder=None)
 
+app_config = {
+    'log_level': 'INFO',
+    'host': '0.0.0.0',
+    'port': 8001,
+    'ws_check_interval': 10,
+    'ws_timeout': 60,
+    'node_rpc_urls': ['ws://127.0.0.1:5556'],
+    'node_max_unsynchronized_block_drift': 0,
+    'node_min_peers': 10,
+}
+
+
+def read_readiness_status():
+    with readiness_status['lock']:
+        return readiness_status['status']
+
+
+def write_readiness_status(status):
+    with readiness_status['lock']:
+        readiness_status['status'] = status
+
+
+def uri_validator(url):
+    try:
+        result = urlparse(url)
+        return all([result.scheme, result.netloc])
+    except:
+        return False
+
+
+def run_error(msg):
+    print('fatal run error! ' + msg, file=sys.stderr)
```

Comment:
why not `logging.error`? Is not initialized yet? 
```suggestion
    logging.error('fatal run error! ' + msg, file=sys.stderr)
```